src/Assignments/ProblemIframe.tsx

```diff
@@ -71,6 +79,79 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         setHeight(`${scrollHeight}px`);
     };
 
+    const formDataToObject = (formData: FormData) => {
+        let object:any = {};
+        // @ts-ignore
+        for(let pair of formData.entries()) {
+            if (_.isUndefined(object[pair[0]])) {
+                object[pair[0]] = pair[1];
+            } else {
+                if(!_.isArray(object[pair[0]])) {
+                    object[pair[0]] = [object[pair[0]]];
+                }
+                object[pair[0]].push(pair[1]);
+            }
+        }
+        return object;
+    };
+
+    async function prepareAndSubmit(problemForm: HTMLFormElement, clickedButton?: HTMLButtonElement) {
+        const submitAction = (window as any).submitAction;
+        if(typeof submitAction === 'function') submitAction(); // this is a global function from renderer - prepares form field for submit
+
+        let formData = new FormData(problemForm);
+        if (_.isNil(problem.grades)) {return;} // just for typescript
+        if (_.isNil(problem.grades[0])) {return;} // not enrolled
```

Comment:
So this isn't an error, since you can submit questions when you are not enrolled

The valid use case for this is a professor trying their own problems

Does this prevent a professor from submitting their answers?